src/app/core/services/users.service.ts

```diff
@@ -65,48 +70,136 @@ export class UsersService {
-      const gameData = await game.ref.get();
-      if (game.in_collection) {
-        collection.games.push({
-          general: {
-            ...gameData.data(),
-            id: game.ref.id,
-          } as GameWithId,
-          user: game,
-        });
-      }
-    }
+    await Promise.all(
+      userData.document.collection.games.map(async (game) => {
+        const gameRef = await game.ref.get();
+        const gameData = gameRef.data();
+        if (gameData && game.in_collection) {
+          collection.games.push({
+            id: gameRef.id,
+            title: gameData.title,
+            platform: gameData.platform ?? '',
+            format: convertFormat(game.format),
+            progress: convertProgress(game.progress),
+            added_on: game.added_on,
+          });
+        }
+      }),
+    );
 
-    for (const movie of userData.document.collection.movies) {
-      const movieData = await movie.ref.get();
-      if (movie.in_collection) {
-        collection.movies.push({
-          general: {
-            ...movieData.data(),
-            id: movie.ref.id,
-          } as MovieWithId,
-          user: movie,
-        });
-      }
-    }
+    await Promise.all(
+      userData.document.collection.movies.map(async (movie) => {
+        const movieRef = await movie.ref.get();
+        const movieData = movieRef.data();
+        if (movieData && movie.in_collection) {
+          collection.movies.push({
+            id: movieRef.id,
+            title: movieData.title,
+            director: movieData.director ?? '',
+            format: convertFormat(movie.format),
+            progress: convertProgress(movie.progress),
+            added_on: movie.added_on,
+          });
+        }
+      }),
+    );
 
-    for (const album of userData.document.collection.albums) {
-      const albumData = await album.ref.get();
-      if (album.in_collection) {
-        collection.albums.push({
-          general: {
-            ...albumData.data(),
-            id: album.ref.id,
-          } as AlbumWithId,
-          user: album,
-        });
-      }
-    }
+    await Promise.all(
+      userData.document.collection.albums.map(async (album) => {
+        const albumRef = await album.ref.get();
+        const albumData = albumRef.data();
+        if (albumData && album.in_collection) {
+          collection.albums.push({
+            id: albumRef.id,
+            title: albumData.title,
+            artist: albumData.artist ?? '',
+            format: convertFormat(album.format),
+            progress: convertProgress(album.progress),
+            added_on: album.added_on,
+          });
+        }
+      }),
+    );
 
     return collection;
   }
 
+  getStatistics(collection: UserCollection) {
+    const userStatistics = {
+      games: statisticsTemplate(),
+      movies: statisticsTemplate(),
+      albums: statisticsTemplate(),
+    };
+
+    collection.games.map((game) => {
```

Comment:
This is a lot of code that ultimately does very little, any ideas how to improve this?